### PR TITLE
Add cachedir-setting and use it for Symfony cache

### LINKED ISSUE #1931: 2.1: Cache being written to temp directory
**Describe the bug**

```
SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Backtrace:
1 public/_include.php:28 (SimpleSAML_exception_handler)
0 [builtin] (N/A)
Caused by: RuntimeException: Unable to write in the "cache" directory (/tmp/simplesaml/cache/saml).
Backtrace:
5 vendor/symfony/http-kernel/Kernel.php:641 (Symfony\Component\HttpKernel\Kernel::buildContainer)
4 vendor/symfony/http-kernel/Kernel.php:545 (Symfony\Component\HttpKernel\Kernel::initializeContainer)
3 vendor/symfony/http-kernel/Kernel.php:787 (Symfony\Component\HttpKernel\Kernel::preBoot)
2 vendor/symfony/http-kernel/Kernel.php:190 (Symfony\Component\HttpKernel\Kernel::handle)
1 src/SimpleSAML/Module.php:234 (SimpleSAML\Module::process)
0 public/module.php:17 (N/A)
```

The /tmp directory is being emptied on most OS'es, so we should be writing Symfony cache elsewhere (like /var/cache)

### PR DESCRIPTION
**Pull request by tvdijen**
Closes #1931

### CODE REVIEW

File: config/config.php.dist [hunk 1]
```diff
@@ -58,6 +58,7 @@ $config = [
     /*
      * The following settings are *filesystem paths* which define where
      * SimpleSAMLphp can find or write the following things:
+     * - 'cachedir': Where Symfony can write it's cache.
```
Comment (thijskh) [inline config/config.php.dist:61]:
```suggestion
     * - 'cachedir': Where SimpleSAMLphp can write its cache. 
```

File: config/config.php.dist [hunk 2]
```diff
@@ -66,9 +67,10 @@ $config = [
      * When specified as a relative path, this is relative to the SimpleSAMLphp
      * root directory.
      */
+    'cachedir' => '/var/cache/simplesamlphp',
     //'loggingdir' => '/var/log/',
     //'datadir' => '/var/data/',
-    'tempdir' => '/tmp/simplesaml',
+    'tempdir' => '/tmp/simplesamlphp',
```
Comment (thijskh) [inline config/config.php.dist:73]: Begs the question whether this is still used - will file a separate issue to discuss that.

**Review by thijskh — COMMENTED** [review]
Probably we need to add install docs that you need to create this dir and make it writable for the webserver...

**Comment by thijskh** [comment]
What is the behaviour if the configured dir is not present and/or not writable for the web server? Does it provide a clear error message?

**Comment by tvdijen** [comment]
The error is as described in the attached issue. It's pretty clear.

I have a setup with php-fpm (running on a dedicated service account per pool) and when the OS cleans up the /tmp-directory I get this error. The strange thing is that when I restart php-fpm, the directory is automatically re-created. I can't explain, but I _do_ know that static cache shouldn't be written to /tmp

File: config/config.php.dist (hunk 1 above)
Comment (thijskh) [reply config/config.php.dist:61]: My comment is more that for the user/deployer the term "Symfony" does not necessarily mean anything. It's just cache for SimpleSAML.

**Comment by cicnavi** [comment]
> The error is as described in the attached issue. It's pretty clear.
> 
> I have a setup with php-fpm (running on a dedicated service account per pool) and when the OS cleans up the /tmp-directory I get this error. The strange thing is that when I restart php-fpm, the directory is automatically re-created. I can't explain, but I _do_ know that static cache shouldn't be written to /tmp

Hi, maybe this is related to having SimpleSAMLphp related cron jobs set up, but which run under different user(s). The first cron job which runs after the reboot, sets the tmp permission... Recently this was the actual case for us.

**Comment by tvdijen** [comment]
Thanks Marco! I suspect is has something to do with opcache. I have no cron jobs set up on my IdPs

**Pull request merged**